File: src/View To Image/Objective-C/View To Image Objc/ViewController.m
//
//  ViewController.m
//  View To Image Objc
//
//  Created by Alex Duffell on 18/12/2017.
//  Copyright © 2017 Alex Duffell. All rights reserved.
//
// Project files are from of my UIView -> UIImage tutorial on wordpress
// https://alexduffell.wordpress.com/2017/12/19/how-to-make-a-uiimage-from-a-uiview-gradients/

#import "ViewController.h"
#import <Foundation/Foundation.h>

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UIProgressView *progressView;
@property (weak, nonatomic) IBOutlet UIView *containerView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _progressView.transform = CGAffineTransformScale(CGAffineTransformIdentity, 1, 20);
    
    _containerView.layer.cornerRadius = 10;
    _containerView.layer.masksToBounds = true;
    
    CAGradientLayer *gradientLayer = [CAGradientLayer layer];
    
    UIColor *primary = [UIColor yellowColor];
    UIColor *highlight = [UIColor redColor];
    
    gradientLayer.frame = _containerView.bounds;
    gradientLayer.colors = @[(id)primary.CGColor, (id)highlight.CGColor];
    gradientLayer.startPoint = CGPointMake(0.0, 0.5);
    gradientLayer.endPoint = CGPointMake(1.0, 0.5);
    
    UIView *gradientView = [[UIView alloc] initWithFrame:_containerView.bounds];
    [gradientView.layer addSublayer:gradientLayer];
    
    UIGraphicsImageRenderer *renderer = [[UIGraphicsImageRenderer alloc] initWithSize:gradientView.bounds.size];
    UIImage *gradientImage = [renderer imageWithActions:^(UIGraphicsImageRendererContext * _Nonnull rendererContext) {
        [gradientView drawViewHierarchyInRect:gradientView.bounds afterScreenUpdates:true];
    }];
    
    _progressView.progressImage = gradientImage;

}




@end

File: src/View To Image/Swift/View To Image Swift/ViewController.swift
//
//  ViewController.swift
//  View To Image Swift
//
//  Created by Alex Duffell on 18/12/2017.
//  Copyright © 2017 Alex Duffell. All rights reserved.
//
// Project files are from of my UIView -> UIImage tutorial on wordpress
// https://alexduffell.wordpress.com/2017/12/19/how-to-make-a-uiimage-from-a-uiview-gradients/

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var progressView: UIProgressView!
    @IBOutlet weak var containerView: UIView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        //Increase height of progress view
        progressView.transform = CGAffineTransform(scaleX: 1, y: 20)
        
        //Round the corners of the progress view (using its container view)
        containerView.layer.cornerRadius = 10
        containerView.layer.masksToBounds = true
        
        //Create a gradient
        let colour1 = UIColor.yellow.cgColor
        let colour2 = UIColor.red.cgColor
        
        let gradientLayer = CAGradientLayer()
        gradientLayer.frame = containerView.bounds
        gradientLayer.colors = [colour1, colour2]
        gradientLayer.startPoint = CGPoint(x: 0.0, y: 0.5)
        gradientLayer.endPoint = CGPoint(x: 1.0, y: 0.5)
        
        //Create a view
        let gradientView = UIView(frame: containerView.bounds)
        gradientView.layer.addSublayer(gradientLayer)
        
        let renderer = UIGraphicsImageRenderer(size: gradientView.bounds.size)
        let image = renderer.image { ctx in
            gradientView.drawHierarchy(in: gradientView.bounds, afterScreenUpdates: true)
        }
        
        progressView.progressImage = image
    }
    
}

